packages/app/src/components/asset-search/asset-search.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useMemo, useState, useEffect } from 'react';
+import { Box, Text, Flex, Input } from '@stacks/ui';
+import { useCombobox } from 'downshift';
+import { useFetchBalances } from '@common/hooks/use-account-info';
+import { getAssetStringParts, toHumanReadableStx, truncateMiddle } from '@stacks/ui-utils';
+import { Asset, searchInputStore, selectedAssetStore } from '@store/recoil/asset-search';
+import { useRecoilState, useRecoilValue, useSetRecoilState } from 'recoil';
+import { LoadingRectangle } from '../loading-rectangle';
+import type { AddressBalanceResponse } from '@blockstack/stacks-blockchain-api-types';
+import { AssetResult } from './asset-search-result';
+import { SelectedAsset } from './selected-asset';
+
+export const AssetSearchField: React.FC = () => {
+  const balancesLoadable = useFetchBalances();
+  const selectedAsset = useRecoilValue(selectedAssetStore);
+  const setSelectedAsset = useSetRecoilState(selectedAssetStore);
+  const [searchInput, setSearchInput] = useRecoilState(searchInputStore);
+  const balancesJSON = JSON.stringify(balancesLoadable.value);
+  const [searchResults, setSearchResults] = useState<Asset[]>([]);
+
+  const assets: Asset[] = useMemo(() => {
+    if (!balancesJSON) return [];
+    const balances: AddressBalanceResponse = JSON.parse(balancesJSON);
```

Comment:
can throw an exception if bad, generally should do this in a wapper

Reply:
This is really just a workaround for `useMemo` not working very well with a bigger js object. I'd prefer to just solve it the right way, but I'm not 100% sure what the cause is, or even what the original problem behavior was.

This is serialized immediately before the hook, so at least there should never be an exception